File: module_project/TabbedApplication/Objective-C/TabbedApplication/MuiltFunctionTabBarController.m
//
//  MuiltFunctionTabBarController.m
//  TabbedApplication
//
//  Created by 小口哲史 on 2016/06/05.
//  Copyright © 2016年 yourCompany. All rights reserved.
//
#import "MuiltFunctionTabBarController.h"
@interface MuiltFunctionTabBarController ()
@end
@implementation MuiltFunctionTabBarController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
    [notificationCenter removeObserver:self];
    [notificationCenter addObserver:self selector:@selector(seguefirst02View) name:LOCAL_NOTIFICATION_SUGUE_FIRST_02_VIEW object:nil];
    [notificationCenter addObserver:self selector:@selector(segueSecond02View) name:LOCAL_NOTIFICATION_SUGUE_SECOND_02_VIEW object:nil];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)seguefirst02View {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    First_02ViewController *first_02ViewCon = [storyboard instantiateViewControllerWithIdentifier:@"First_02ViewController"];
    
    self.selectedIndex = 0;
    [(UINavigationController*)[self.viewControllers objectAtIndex:0]
     pushViewController:first_02ViewCon animated:YES];
}

-(void)segueSecond02View {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    Second_02ViewController *second_02ViewCon = [storyboard instantiateViewControllerWithIdentifier:@"Second_02ViewController"];
    self.selectedIndex = 1;
    [(UINavigationController*)[self.viewControllers objectAtIndex:1]
     pushViewController:second_02ViewCon animated:YES];
}

@end

File: module_project/TabbedApplication/Swift/TabbedApplication/MuiltFunctionTabBarController.swift
//
//  MuiltFunctionTabBarController.swift
//  TabbedApplication
//
//  Created by 小口 哲史 on 2017/05/17.
//  Copyright © 2017年 小口 哲史. All rights reserved.
//

import UIKit

class MuiltFunctionTabBarController: UITabBarController {
    override func viewDidLoad() {
        super.viewDidLoad()
        
        //タブバーカスタマイズ
        tabBar.items![2].image = UIImage(named: "third.png")!.withRenderingMode(UIImageRenderingMode.alwaysTemplate)
        tabBar.items![2].title = "third"
        //tabBar.items![2].selectedImage = UIImage(named: "third.png")
        UITabBar.appearance().tintColor = UIColor.red
        UITabBar.appearance().barTintColor = UIColor.blue
        
        let nc:NotificationCenter = NotificationCenter.default
        nc.removeObserver(self)
        nc.addObserver(self, selector: #selector(self.segueFirst02View), name: NSNotification.Name(rawValue: LOCAL_NOTIFICATION_SUGUE_FIRST_02_VIEW), object: nil)
        nc.addObserver(self, selector: #selector(self.segueFirstRootView), name:NSNotification.Name(rawValue: LOCAL_NOTIFICATION_SUGUE_FIRST_ROOT_VIEW), object: nil)
        nc.addObserver(self, selector: #selector(self.segueSecond02View), name: NSNotification.Name(rawValue:LOCAL_NOTIFICATION_SUGUE_SECOND_02_VIEW), object: nil)
        nc.addObserver(self, selector: #selector(self.segueSecondRootView), name: NSNotification.Name(rawValue:LOCAL_NOTIFICATION_SUGUE_SECOND_ROOT_VIEW), object: nil)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func segueFirst02View(){
        let storyboard: UIStoryboard = UIStoryboard(name:"Main", bundle:nil)
        let vc = storyboard.instantiateViewController(withIdentifier: "First_02ViewController") as! First_02ViewController
        selectedIndex = 0
        (viewControllers?[0] as? UINavigationController)?.pushViewController(vc, animated: true)
    }
    func segueFirstRootView() {
        selectedIndex = 0
        let nvc:UINavigationController = (viewControllers?[0] as? UINavigationController)!
        nvc.popToRootViewController(animated: true)
    }
    func segueSecond02View() {
        let storybord:UIStoryboard = UIStoryboard(name:"Main",bundle:nil)
        let vc = storybord.instantiateViewController(withIdentifier:"Second_02ViewController") as! Second_02ViewController
        selectedIndex = 1
        (viewControllers?[1] as? UINavigationController)?.pushViewController(vc, animated: true)
    }
    func segueSecondRootView() {
        selectedIndex = 1
        let nvc:UINavigationController = (viewControllers?[1] as? UINavigationController)!
        nvc.popToRootViewController(animated: true)
    }
}
